File: packages/gbnf-experiment/data/20260909T194429Z_a7bb3ecf/ported_implementation/rules_builder/parse_space.py
from __future__ import annotations

from ..utils.js import char_at


def parse_space(src: str, pos: int, newline_ok: bool) -> int:
    while char_at(src, pos) in (' ', '\t', '#') or (
        newline_ok and char_at(src, pos) in ('\r', '\n')
    ):
        if char_at(src, pos) == '#':
            while char_at(src, pos) and char_at(src, pos) not in ('\r', '\n'):
                pos += 1
        else:
            pos += 1
    return pos

File: proposed-projects/round-trip-experiment/reverse/20260909T190508Z_7c1544c0/20260910T103012Z_7e3d5183/ported_implementation/src/rules-builder/parse-space.ts
import { charAt } from '../utils/js.js';

const isSpace = (char: string, newlineOk: boolean): boolean =>
  char === ' ' ||
  char === '\t' ||
  char === '#' ||
  (newlineOk && (char === '\r' || char === '\n'));

export const parseSpace = (src: string, pos: number, newlineOk: boolean): number => {
  while (isSpace(charAt(src, pos), newlineOk)) {
    if (charAt(src, pos) === '#') {
      let char = charAt(src, pos);
      while (char && char !== '\r' && char !== '\n') {
        pos += 1;
        char = charAt(src, pos);
      }
    } else {
      pos += 1;
    }
  }
  return pos;
};
